### ISSUE TITLE
clang should warn the use of statement expression which is a GNU extension

### ISSUE THREAD

**Issue by usx95** [llvm/llvm-project#90852]
```cpp
struct Printy {
    Printy(int);
    Printy(const Printy&);
    ~Printy();
};

void Foo() {
    Printy P[] = {Printy(1), ({
                      int x = 2;
                      Printy p(x);
                      p;
                  })};
}
```
This contains statement-expression which is a GNU extension. Although Clang supports stmt-exprs to a large extent, Clang should still warn against the use of this non-standard extension.

**Comment by usx95** [comment]
Oops. I might have filed the issue without looking at the prior work.
It looks like this is already implemented as `-Wgnu-statement-expression`. It is just that the warning is not enabled by default (which is also reasonable).
